src/wing/CCPACSWingComponentSegment.h

```diff
@@ -51,7 +56,7 @@ typedef std::vector<CCPACSWingSegment*>       SegmentList;
 
 class CCPACSWing;
 
-class CCPACSWingComponentSegment : public CTiglAbstractSegment
+class CCPACSWingComponentSegment : public CTiglAbstractSegment, public CCPACSWingComponentSegmentLineFunctionality
```

Comment:
By Subclassing from CCPACSWingComponentSegmentLineFunctionality, you still blow up the API from WingComponentSegment. This does not improve the situation compared to the previous version, it just looks cleaner on the first sight.

Better would be a completely separate Class for CCPACSWingComponentSegmentLineFunctionality, that uses the ComponentSegment as a member or input for its computations.